controllers/octavia_controller.go

```diff
@@ -609,6 +610,23 @@ func (r *OctaviaReconciler) reconcileNormal(ctx context.Context, instance *octav
 			err.Error()))
 		return ctrl.Result{}, err
 	}
+	apiObsGen, err := r.checkOctaviaAPIGeneration(instance)
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			octaviav1.OctaviaAPIReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			octaviav1.OctaviaAPIReadyErrorMessage,
+			err.Error()))
+		return ctrlResult, nil
+	}
+	if !apiObsGen {
+		instance.Status.Conditions.Set(condition.UnknownCondition(
+			octaviav1.OctaviaAPIReadyCondition,
+			condition.InitReason,
+			octaviav1.OctaviaAPIReadyInitMessage,
+		))
+	}
```

Comment:
we should not mirror the ready count and conditions from the api if it's observedGeneration is not up to date.

Reply:
isn't sufficient to mark the condition as `not ready` (Unknown) and Mirror? In theory `Mirror` shouldn't be required at all because we have it in the `defer` function anyway.
In addition, looking at this code I'm not sure about L641:

```golang
...
} else {
instance.Status.Conditions.MarkTrue(octaviav1.OctaviaAPIReadyCondition, condition.DeploymentReadyMessage)
}
```

What about removing the `Mirror(` ? Do you see any drawback? (e.g., introduce regressions in the current code?)

Reply:
we we set the condition to unknown then mirror the condition from the API then the mirror will override the unknown state we set before based on the API's conditions. 

Btw the mirror call in the defer function is not the same call that the mirror here. Here we mirror the summary(i.e. the values of the most severe and most recent condition) of the conditions from the OctaviaAPI.Status.Conditions into the Octavia.Status.condition[OctaviaAPIReadyCondition]. In the defer function we user mirror make sure that the Octavia.Status.Condition[ReadyCondition] is up to date based on the rest of the conditions in Octavia.Status.Conditions.

Reply:
What I would do here is to put the mirroring and the readycount copying under an if that checks that the observedGeneration of the API is up to date.

Reply:
You're right about the Mirroring, we need to mirror conditions form the underlying CR, so that block should stay, I totally agree, we have the same logic in Manila to set the `ManilaAPIReadyCondition`. Also, it doesn't make sense mirror a condition from a `Generation` that is not the current one, so we might have something like the following:

```golang
if !apiObsGen {
	instance.Status.Conditions.Set(condition.UnknownCondition(
		octaviav1.OctaviaAPIReadyCondition,
		condition.InitReason,
		octaviav1.OctaviaAPIReadyInitMessage,
	))
} else {
	// Mirror OctaviaAPI status' ReadyCount to this parent CR
	// TODO(beagles): We need to have a way to aggregate conditions from the other services into this
	//
	instance.Status.OctaviaAPIReadyCount = octaviaAPI.Status.ReadyCount
	conditionStatus := octaviaAPI.Status.Conditions.Mirror(octaviav1.OctaviaAPIReadyCondition)
	if conditionStatus != nil {
		instance.Status.Conditions.Set(conditionStatus)
	}
}
```

With the above, we only Set the Condition to `UnknowCondition` if it's not the current `ObservedGeneration`, otherwise we can mirror whatever comes from the underlying level.
Are we on the same page on this view?

Reply:
Done, we only `mirror` when we see the last generation.